Lib/test/test_ctypes/test_dlerror.py

```diff
@@ -0,0 +1,114 @@
+import os
+import sys
+import unittest
+
+FOO_C = r"""
+#include <unistd.h>
+
+/* This is a 'GNU indirect function' (IFUNC) that will be called by
+   dlsym() to resolve the symbol "foo" to an address. Typically, such
+   a function would return the address of an actual function, but it
+   can also just return NULL.  For some background on IFUNCs, see
+   https://willnewton.name/uncategorized/using-gnu-indirect-functions.
+
+   Adapted from Michael Kerrisk's answer: https://stackoverflow.com/a/53590014.
+*/
+
+asm (".type foo, @gnu_indirect_function");
```

Comment:
`asm` syntax is architecture-specific; the [cross-platform but GCC-only way to spell this](https://www.sourceware.org/binutils/docs/as/Type.html#Type) is:
```suggestion
asm (".type foo, STT_GNU_IFUNC");
```

Reply:
From the documentation you linked:
```
The syntaxes supported are:

  .type <name> STT_<TYPE_IN_UPPER_CASE>
  [...]
  .type <name>,@<type>
  [...]
```

Therefore the two choices we have are:

1. `.type foo, @gnu_indirect_function`
2. `.type foo STT_GNU_IFUNC`

Notice the lack of comma in (2).

Your commit suggestion offers a hybrid variant, which strangely still works.

Nevertheless, these two seem equivalent to me. What makes the `STT_GNU_IFUNC` variant cross-platform?

Reply:
Oh, thanks for the catch!
```suggestion
asm (".type foo STT_GNU_IFUNC");
```

The docs also say:

> Because some of the characters used in these syntaxes (such as ‘@’ and ‘#’) are comment characters for some architectures, some of the syntaxes below do not work on all architectures. The first variant will be accepted by the GNU assembler on all architectures so that variant should be used for maximum portability, if you do not need to assemble your code with other assemblers.

Here's the failure from a Raspberry Pi: https://buildbot.python.org/api/v2/logs/11673079/raw_inline

Reply:
Ah I see, so we both fixed something here :stuck_out_tongue: